DICOMPlugins/DICOMTID1500Plugin.py

```diff
@@ -124,6 +148,114 @@ def createLoadableAndAddReferences(self, datasets):
+      #     SOPInstanceUIDs = list(set(SOPInstanceUIDs))
+      #     # Now add to the loadable.referencedInstanceUIDs 
+      #     if (SOPInstanceUIDs):
+      #       loadable.referencedInstanceUIDs += SOPInstanceUIDs 
+          
+        # We also check for FrameOfReferenceUID, and add to the list of loadable.referencedInstanceUIDs 
+        if (loadable.ReferencedSegmentationInstanceUIDs[uid] == []) and (loadable.referencedInstanceUIDs == []): 
+          print('Check for FrameOfReferenceUID')
+          FrameOfReferenceUIDs = [] 
+          # create the ImageRegion3D code 
+          image_region_code = hd.sr.value_types.Code(
+              value='111030',
+              scheme_designator='DCM',
+              meaning='Image Region'
+          )
+          # First get the planar roi measurement groups 
+          groups = sr.content.get_planar_roi_measurement_groups()
+          # Then we check if the SR contains a point or not 
+          # Iterate through the groups 
+          for group in groups: 
+            # Check if there is a reference_type, should be ImageRegion
+            try: 
+              reference_type = group.reference_type
+            except: 
+              print('reference_type does not exist for group')
+            # If it does equal image_region_code, then check if it's a POINT
+            if (reference_type == image_region_code):
+              try: 
+                graphic_type = group.roi.GraphicType
+              except: 
+                print('GraphicType does not exist for group.roi')
+              if (graphic_type == "POINT"): 
+                FrameOfReferenceUIDs.append(group.roi.ReferencedFrameOfReferenceUID)
+            # unique FrameOfReferenceUIDs 
+            FrameOfReferenceUIDs = list(set(FrameOfReferenceUIDs))
+            # Now we get the SeriesInstanceUIDs in the dicom database that have these FrameOfReferenceUIDs 
+            # We know that the StudyInstanceUID must be the same, so we only check those patients. 
+            StudyInstanceUID = dataset.StudyInstanceUID 
+            # Get the list of series in the database for this study 
+            SeriesInstanceUIDs = slicer.dicomDatabase.seriesForStudy(StudyInstanceUID)
+            # Now get the FrameOfReferenceUIDs for these SeriesInstanceUIDs 
+            possible_SeriesInstanceUIDs = [] 
+            for SeriesInstanceUID in SeriesInstanceUIDs: 
+              FrameOfReferenceUID_forSeries = slicer.dicomDatabase.fileValue(
+                                                        slicer.dicomDatabase.filesForSeries(SeriesInstanceUID)[0], "0020,0052")
+              # check if this matches any of the FrameOfReferenceUIDs, if so, add to list of possible_SeriesInstanceUIDs  
+              if FrameOfReferenceUID_forSeries in FrameOfReferenceUIDs: 
+                possible_SeriesInstanceUIDs.append(SeriesInstanceUID)
+            # Now get all the instances for these possible_FrameOfReferenceUIDs 
```

Comment:
This section of code handles `POINT`, which should reference specific images. Collecting series by `FrameOfReferenceUID` should only be done for `POINT3D`!